src/core/pagerankGraph.test.js

```diff
@@ -165,6 +166,131 @@ describe("core/pagerankGraph", () => {
     });
   });
 
+  describe("edge filtering", () => {
+    const src1 = NodeAddress.fromParts(["src", "1"]);
+    const src2 = NodeAddress.fromParts(["src", "2"]);
+    const dst1 = NodeAddress.fromParts(["dst", "1"]);
+    const dst2 = NodeAddress.fromParts(["dst", "2"]);
+    const e11 = {
+      src: src1,
+      dst: dst1,
+      address: EdgeAddress.fromParts(["e", "1", "1"]),
+    };
+    const e12 = {
+      src: src1,
+      dst: dst2,
+      address: EdgeAddress.fromParts(["e", "1", "2"]),
+    };
+    const e21 = {
+      src: src2,
+      dst: dst1,
+      address: EdgeAddress.fromParts(["e", "2", "1"]),
+    };
+    const e22 = {
+      src: src2,
+      dst: dst2,
+      address: EdgeAddress.fromParts(["e", "2", "2"]),
+    };
+    const graph = () =>
+      [e11, e12, e21, e22].reduce(
+        (g, e) => g.addEdge(e),
+        [src1, src2, dst1, dst2].reduce((g, n) => g.addNode(n), new Graph())
+      );
+
+    const pagerankGraph = () => new PagerankGraph(graph(), defaultEvaluator);
+
+    function expectPagerankGraphToEqualGraph(options: EdgesOptions | void) {
+      const pagerankGraphEdges = Array.from(
+        pagerankGraph().edges(options)
+      ).sort();
+      const graphEdges = Array.from(graph().edges(options)).sort();
+
+      const edgesAreEqual = (e1, e2) =>
```

Comment:
We don't need to define functions like this, since we can just import "lodash.isequal". We usually import it as `deepEqual`, so `git grep deepEqual` will find you examples.

Also, be aware that expect(object).toEqual(object) is a thing, and also uses deep-equal semantics

Reply:
Thanks! Remembering now that William [spoke on this](https://www.youtube.com/watch?v=xKfIOcN-xnA#t=1h47m21s) at live coding pgGraph.